test/zIntegrationTests.ts

```diff
@@ -1,48 +1,131 @@
-import { helperResetNetwork, helperSwapETHWithOUSD } from "./MainnetHelper";
-import { ethers } from "hardhat";
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { parseEther, formatEther } from "ethers/lib/utils";
 import { expect } from "chai";
 import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
+import { helperResetNetwork, helperSwapETHWithOUSD, defaultBlockNumber, helperSwapETHWith3CRV } from "./MainnetHelper";
+import { createAndFundMetapool, fundMetapool } from "./CurveHelper";
 
-/* Integration tests start here */
+let r: ContractTestContext;
+let owner: SignerWithAddress;
+let user: SignerWithAddress;
+let pretenderOUSDSigner : SignerWithAddress;
+let lvUSD3CRVPoolInstance;
 
-let contractARCHToken;
+const userOUSDPrinciple = 100;
+const initialFundsInPool = 700;
+const initialCoordinatorLvUSDBalance = 10000;
 
-describe("Setting the stage: Getting some OUSD and deploying our contracts", function () {
-    let signer;
-    let r: ContractTestContext;
-    let lvUSD;
+let adminInitial3CRVBalance: number;
 
+function parseEtherNum (num) {
+    return parseEther(num.toString());
+}
+
+function getFloatFromBigNum (bigNumValue) {
+    return parseFloat(formatEther(bigNumValue));
+}
+
+async function printPoolState (poolInstance) {
+    console.log(
+        "Pool has %s coin0/lvUSD and %s coin1/3CRV",
+        getFloatFromBigNum(await poolInstance.balances(0)),
+        getFloatFromBigNum(await poolInstance.balances(1)),
+    );
+}
+
+async function setupEnvForIntegrationTests () {
+    // Reset network before integration tests
+    await helperResetNetwork(defaultBlockNumber);
+
+    // Setup & deploy contracts
+    r = await buildContractTestContext();
+    owner = r.owner;
+    user = r.addr1;
+    pretenderOUSDSigner = r.addr3;
+
+    /* ====== Setup accounts and funds ===========
+    expected state:
+    - admin has 1000 lvUSD and 1000 3CRV tokens, to fund pool
+    - User has 100 OUSD to use as principle
+    - pretenderOUSDSigner, which will act as OUSD rebase agent, has 500 OUSD
+    - Coordinator gets initialCoordinatorLvUSDBalance of lvUSD so it can use it when getting leverage.
+    */
+
+    // Prep owner accounts with funds needed to fund pool
+    await r.lvUSD.mint(await owner.getAddress(), parseEther("1000.0"));
```

Comment:
Change to `parseUnits`? I think we may want to start using this instead of parseEther to avoid confusion with what the number actually represents. We aren't minting the equivalent of 1000 ether of lvUSD but this always makes me think that when I read it

Reply:
done